source/kernel/src/daemons/shells.rs

```diff
@@ -222,94 +227,129 @@ pub async fn graphical_shell_mono(k: &'static Kernel, settings: GraphicalShellSe
     })
     .await;
 
-    let mut any = true;
-
-    let interval = Duration::from_secs(1) / 20;
-
     loop {
-        if any {
-            ring_drawer::drawer_bw(&mut fc_0, &rline, style.clone()).unwrap();
-            fc_0 = disp_hdl.draw_mono(fc_0).await.unwrap();
-            any = false;
-        }
+        // Draw to the display
+        ring_drawer::drawer_bw(&mut fc_0, &rline, style.clone()).unwrap();
+        fc_0 = disp_hdl.draw_mono(fc_0).await.unwrap();
+
+        // Poll ONCE until there is progress, with unlimited time
+        io_poll(PollStyle::OneShot, &mut keyboard, &mut rline, &tid_io).await;
 
+        // SOMETHING happened, so now try and grab as many things as possible
+        // until the debounce timer expires
         let _ = k
             .timeout(
-                interval,
-                io_poll(&mut any, &mut keyboard, &mut rline, &tid_io),
+                redraw_debounce,
+                io_poll(PollStyle::Forever, &mut keyboard, &mut rline, &tid_io),
```

Comment:
nit, take it or leave it: would it be simpler to just make this
```suggestion
                async move {
                    loop {
                        io_poll(&mut keyboard, &mut rline, &tid_io),
                    }
                 }
```
instead of passing the poll style into the `io_poll` function? or was there some reason that didn't work?

Reply:
None, other than a vague preference to use async functions over blocks.